internal/fs/file_windows.go

```diff
@@ -61,25 +33,40 @@ func fixpath(name string) string {
 	return name
 }
 
-// TempFile creates a temporary file.
+// TempFile creates a temporary file which is marked as delete-on-close
 func TempFile(dir, prefix string) (f *os.File, err error) {
+	// slightly modified implementation of ioutil.TempFile(dir, prefix) to allow us to add
+	// the FILE_ATTRIBUTE_TEMPORARY | FILE_FLAG_DELETE_ON_CLOSE flags.
+	// These provide two large benefits:
+	// FILE_ATTRIBUTE_TEMPORARY tells Windows to keep the file in memory only if possible
+	// which reduces the amount of unnecessary disk writes.
+	// FILE_FLAG_DELETE_ON_CLOSE instructs Windows to automatically delete the file once
+	// all file descriptors are closed.
+
 	if dir == "" {
 		dir = os.TempDir()
 	}
 
-	access := uint32(syscall.GENERIC_READ | syscall.GENERIC_WRITE)
-	creation := uint32(syscall.CREATE_NEW)
-	flags := uint32(FILE_ATTRIBUTE_TEMPORARY | FILE_FLAG_DELETE_ON_CLOSE)
-	
+	access := uint32(windows.GENERIC_READ | windows.GENERIC_WRITE)
+	creation := uint32(windows.CREATE_NEW)
+	share := uint32(0) // prevent other processes from accessing the file
+	flags := uint32(windows.FILE_ATTRIBUTE_TEMPORARY | windows.FILE_FLAG_DELETE_ON_CLOSE)
+
 	for i := 0; i < 10000; i++ {
-		path := filepath.Join(dir, prefix+nextRandom())
+		randSuffix := strconv.Itoa(int(1e9 + rand.Intn(1e9)%1e9))[1:]
```

Comment:
I think it's a bad idea to use `rand.Intn()` from the `math` package since it's deterministic: https://go.dev/play/p/ofccaFGv1mc so two different restic processes will generate exactly the same names for the temp files. Or am I wrong?

Reply:
Oh, I've overlooked that. Now the code creates a new random number generator instance and seeds it with the current time in nanoseconds. That should be sufficient?

Reply:
Fine with me :)